modules/021-cni-cilium/templates/configmap.yaml

```diff
@@ -27,17 +27,21 @@ data:
 
   bpf-lb-bypass-fib-lookup: "false" # TODO: https://docs.cilium.io/en/v1.11/gettingstarted/kubeproxy-free/#nodeport-with-fhrp-vpc
 
-  tunnel: {{ .Values.cniCilium.tunnelMode | lower | quote }}
-
+  {{- if eq .Values.cniCilium.internal.mode "VXLAN" }}
```

Comment:
NIT: In my opinion it is better to avoid nested clauses and rewrite the code like
```yaml
{{ if eq .Values.cniCilium.internal.mode "VXLAN" }}
tunnel: "vxlan"
enable-bpf-masquarade: "true"
enable-ipv4-egress-gateway: "true"
{{ else if eq .Values.cniCilium.internal.mode "DirectWithNodeRoutes" }}
tunnel: "disabled"
auto-direct-node-routes: "true"
{{ else if eq .Values.cniCilium.internal.mode "Direct" }}
tunnel: "disabled"
{{ else }}
  {{ fail (printf "unknown mode %s" .Values.cniCilium.internal.mode) }}
{{ end }}
```

Reply:
That is how we can see all mode related settings in one place.

Reply:
looks good, i use your code )